chiselFrontend/src/main/scala/chisel3/core/MultiClock.scala

```diff
@@ -0,0 +1,25 @@
+// See LICENSE for license details.
+
+package chisel3.core
+
+import scala.language.experimental.macros
+
+import chisel3.internal._
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+import chisel3.internal.sourceinfo.{SourceInfo}
+
+private[chisel3] final case class ClockAndReset(clock: Clock, reset: Bool)
+
+object withClockAndReset {  // scalastyle:ignore object.name
+  def apply[T](clock: Clock, reset: Bool)(block: => T)(implicit sourceInfo: SourceInfo): T = {
```

Comment:
I'm fine with `.implicitClock` and `.implicitReset`. Also keep in mind we're not talking about overloading the IO signals functionality because this is object `Module.clock`, ie. inside of a given `Module`, `this.clock` or just `clock` returns the input port `clock` to that `Module` while `Module.clock` (or `Module.implicitClock`) would return the actual current implicit clock set by the most recent `withClock`. 

Any easy confusion here though might be reason enough to not call them `.clock` and `.reset`